react-validated-input-component/my-app/src/ValidatedInput.tsx

```diff
@@ -0,0 +1,70 @@
+import { ChangeEvent, useState, KeyboardEvent } from 'react';
+
+export default function ValidatedInput() {
+  const [password, setPassword] = useState('');
+  const [errorMessage, setErrorMessage] = useState({
+    message: '',
+    displayStyle: '',
+    color: '',
+  });
+
+  function handleOnChange(event: ChangeEvent<HTMLInputElement>) {
+    setPassword((prevPassword) => {
+      return event.target.value;
+    });
+
+    const newPassword = event.target.value;
+
+    setErrorMessage((prevErrorMessage) => {
+      if (newPassword.length === 0) {
+        return {
+          ...prevErrorMessage,
+          message: 'A password is required.',
+          displayStyle: 'fa-solid fa-x',
+          color: 'red',
+        };
+      } else if (newPassword.length >= 1 && newPassword.length < 8) {
+        return {
+          ...prevErrorMessage,
+          message: 'Your password is too short.',
+          displayStyle: 'fa-solid fa-x',
+          color: 'red',
+        };
+      } else {
+        return {
+          ...prevErrorMessage,
+          message: '',
+          displayStyle: 'fa-solid fa-check',
+          color: 'green',
+        };
+      }
+    });
+  }
+
+  function handleKeyDown(event: KeyboardEvent<HTMLInputElement>) {
+    if (event.key === 'Enter' && event.currentTarget.type === 'password') {
+      event.preventDefault();
+    }
+  }
```

Comment:
I'd suggest tying this to the form's `'submit'` event whenever there's a form wrapping an input.

Reply:
@TimelikeClosure updated